raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -339,26 +366,42 @@ private VoteResponseData handleVoteRequest(VoteRequestData request) throws IOExc
         }
     }
 
-    private void handleVoteResponse(int remoteNodeId, VoteResponseData response) throws IOException {
-        if (quorum.isLeader()) {
-            logger.debug("Ignoring vote response {} since we already became leader for epoch {}",
+    private boolean handleVoteResponse(
+        RaftResponse.Inbound responseMetadata
+    ) throws IOException {
+        int remoteNodeId = responseMetadata.sourceId();
```

Comment:
As we discussed earlier today, it makes sense to consider introducing some inheritance property into the auto generated protocol for writing less boiler-template code.

Reply:
Yes. This would be helpful in the case of the error code. But it's probably not a blocking feature gap.